pkg/k8s/watchers/pod.go

```diff
@@ -272,20 +280,19 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		return k.deleteK8sPodV1(newK8sPod)
 	}
 
-	logger := log.WithFields(logrus.Fields{
-		logfields.K8sPodName:   newK8sPod.ObjectMeta.Name,
-		logfields.K8sNamespace: newK8sPod.ObjectMeta.Namespace,
-	})
-
-	if oldK8sPod.Spec.HostNetwork || newK8sPod.Spec.HostNetwork {
+	if newK8sPod.Spec.HostNetwork {
 		logger.Debug("Pod is using host networking")
 		return nil
 	}
 
-	// The pod IP can never change, it can only switch from unassigned to
-	// assigned
-	// Process IP updates
-	k.addK8sPodV1(newK8sPod)
+	oldPodIPs := k8sUtils.ValidIPs(oldK8sPod.Status)
+	newPodIPs := k8sUtils.ValidIPs(newK8sPod.Status)
+	err := k.updatePodHostData(oldK8sPod, newK8sPod, oldPodIPs, newPodIPs)
+
+	if err != nil {
+		logger.WithError(err).Warning("Unable to update ipcache map entry on pod update")
+		return err
+	}
```

Comment:
```suggestion
       switch {
       case err != nil:
               msg := "Unable to update ipcache map entry on pod add"
               if err == errIPCacheOwnedByNonK8s {
                       logger.WithError(err).Debug(msg)
               } else {
                       logger.WithError(err).Warning(msg)
               }
       default:
               logger.Debug("Updated ipcache map entry on pod add")
       }
```

Reply:
also add this in line 115
```golang
	errIPCacheOwnedByNonK8s = fmt.Errorf("ipcache entry owned by kvstore or agent")
```

Reply:
First change made. Second change does not seem to be needed as `errIPCacheOwnedByNonK8s` is already defined at `pkg/k8s/watchers/watcher.go:110`.